README.md

```diff
@@ -131,15 +141,397 @@ class Admin < Dill::Role
 end
 ```
 
-##To Be Continued...
+That's it for the walkthrough, but Dill has a lot more to offer. We recommend digging into Widgets next, as Widget is the fundamental class common to all the Dill elements.
+
+
+#Widgets
+A widget is the fundamental Dill element. A widget abstracts a DOM element, allowing you to call methods on that element, like checking for text or submitting a form.
+
+
+## Widget Declaration
+A widget can be declared with the `widget` macro, or as a class.
+
+Macro widget declarations take a CSS or XPath selector as the first argument.
+
+```ruby
+  widget :todo_item, '.todo-item'
+  widget :todo_item, [:xpath, '//some/node']
```

Comment:
this is a valid xpath selector :)

#protip you can test xpath selectors in chrome console with `$x("")`